mula/scheduler/connectors/services/services.py

```diff
@@ -206,11 +208,12 @@ def _verify_response(self, response: httpx.Response) -> None:
         try:
             response.raise_for_status()
         except HTTPError as e:
-            self.logger.error(
+            self.logger.warning(
                 "Received bad response from %s.",
```

Comment:
Yeah I agree, I think this function is probably an artifact as a log wrapper around `raise_for_status()` . Think we can replace the uses of `self._verify_response(response)` with `response.raise_for_status()` in this file. Since, like you mentioned 404 doesn't mean it is a bad response, and let the callers handle and log the exceptions raised by httpx.